### PR TITLE
Run CI tasks on push to main/metrics

### LINKED ISSUE #2083: Linux CI Task Badge
As mentioned on the SIG call today, the Linux badge indicates the task is failing, but it isn't immediately clear why. I think that the problem is that the Linux task isn't actually running when PRs merge. If you click on the ❌ or ✔️ on [any given commit](https://github.com/open-telemetry/opentelemetry-dotnet/commits/main), you'll see the status of 14 tasks, compared to 28 or 29 on a PR.

I haven't been watching the badge so I don't know the case, but if it is occasionally ✔️ and not always ❌ then I would speculate we're seeing the general status of the last run rather than any specific task.

The solution is likely to update the triggers on the task definitions - note the difference between [linux-ci](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/.github/workflows/linux-ci.yml#L3) and [integration](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/.github/workflows/integration.yml#L3). I can submit a PR for this change if you specify whether the change should apply to all tasks or just a subset

### PR DESCRIPTION
**Pull request by TimHess**
Fixes #2083

I did not adjust the trigger for the API Compatibility check as that probably won't be helpful until #2081 moves forward

### CODE REVIEW

**Review by reyang — APPROVED** [review]
LGTM, let's see if it solves the mystery Linux badge issue.

**Review by cijothomas — APPROVED** [review]

**Comment by cijothomas** [comment]
image
We are green :)

**Comment by reyang** [comment]
Mystery solved. THANK YOU @TimHess!

File: .github/workflows/markdownlint.yml [hunk 1]
```diff
@@ -1,6 +1,10 @@
 name: markdownlint
 
 on:
+  push:
+    branches: [ main, metrics ]
+    paths-ignore:
+    - '**.md'
```
Comment (mic-max) [inline .github/workflows/markdownlint.yml:7]: Why is `*.md` being ignored here? Isn't this the exact type of file we want to trigger a `markdownlint` job when they change?

Or is there a reason it only runs on `pull_request` and not on `push`?

Comment (reyang) [reply .github/workflows/markdownlint.yml:7]: This looks like a copypasta, I think it should be `paths` instead of `paths-ignore`.

Comment (cijothomas) [reply .github/workflows/markdownlint.yml:7]: @mic-max Do you want to submit a PR fixing the typo?

**Pull request merged**